main.go

```diff
@@ -287,8 +312,8 @@ func buildFunction(commandString string, rawOutput, rawInput bool) interface{} {
 	if rawInput && rawOutput {
 		f = func(args []interface{}, stdin string) (string, error) {
 			stringArgs, err := castToStrings(args)
-			commandArray = append(commandArray, stringArgs...)
-			command := exec.Command(commandArray[0], commandArray[1:]...)
+			extendedCommandArray := append(commandArray, stringArgs...)
```

Comment:
I found a bug here. The returned function `f` is a closure over `commandArray`. We evaluate `buildFunction` once for the arguments passed in, but we invoke `f` for every invocation.

This means every time you invoke `f` you end up appending to commandArray. So your command ends up being this gigantic aggregate of all your arguments.

Obviously I fixed this by declaring the command array as a new array inside `f`.

Reply:
Err, whoops. Thanks!